pkg/bgp/speaker/speaker.go

```diff
@@ -277,65 +279,46 @@ func (s *MetalLBSpeaker) OnUpdateEndpointSliceV1Beta1(eps *slim_discover_v1beta1
 	return nil
 }
 
-// OnAddNode notifies the Speaker of a new node.
-func (s *MetalLBSpeaker) OnAddNode(node *v1.Node) error {
+// reconcileNode notifies the speaker of a node (K8s Node or CiliumNode) event
+func (s *MetalLBSpeaker) reconcileNode(op Op, nodeMeta *metav1.ObjectMeta, podCIDRs *[]string, withDraw bool) error {
 	if s.shutDown() {
 		return ErrShutDown
 	}
-	if node.GetName() != nodetypes.GetName() {
+	if nodeMeta == nil || nodeMeta.GetName() != nodetypes.GetName() {
```

Comment:
reviewer note: This nil check is added to safeguard against panics. I don't expect this to be hit in practice, but the unit tests will pass `nil` nodes to the node event handlers.